core/scm.js

```diff
@@ -68,8 +57,54 @@ const SCHEMA_COMMIT = Joi.object().keys({
         .example('https://github.com/scredriver-cd/screwdriver/commit/8843d7f92416211de')
 }).label('SCM Commit');
 
+const SCHEMA_HOOK = Joi.object().keys({
+    type: Joi.string()
+        .valid(['pr', 'repo'])
+        .required()
+        .label('Type of the event'),
+
+    action: Joi.alternatives()
+        .when('type', { is: 'pr', then: Joi.valid(['opened', 'closed']) })
+        .when('type', { is: 'repo', then: Joi.valid('push') })
+        .required()
+        .label('Action of the event'),
+
+    prNumber: Joi.number()
+        .integer()
+        .positive()
+        .optional()
+        .label('PR number'),
+
+    checkoutUrl: Joi
+        .string().regex(Regex.CHECKOUT_URL)
+        .required()
+        .label('Checkout URL for the application')
+        .example('git@github.com:screwdriver-cd/data-schema.git#master')
+        .example('https://github.com/screwdriver-cd/data-schema.git#master'),
+
+    branch: Joi.string()
+        .required()
+        .label('Branch of the repository'),
+
+    prRef: Joi.string()
+        .optional()
+        .label('PR reference of the repository'),
+
+    sha: Joi.string().hex()
+        .length(40)
+        .required()
+        .label('Commit SHA')
+        .example('ccc49349d3cffbd12ea9e3d41521480b4aa5de5f'),
+
+    username: Joi.string()
+        .required()
+        .label('Display Name')
```

Comment:
This should be username, not Display Name. They are different.
d2lam vs Dao Lam